client/src/reducers/controls.js

```diff
@@ -476,13 +482,15 @@ const Controls = (
 
       // update the filter to match all selected options
       const cat = newCategoricalSelectionState[action.metadataField];
-      state.dimensionMap[obsAnnoDimensionName(action.metadataField)].filterEnum(
-        ControlsHelpers.selectedValuesForCategory(cat)
-      );
-
+      const dName = obsAnnoDimensionName(action.metadataField);
```

Comment:
Since category metadata selected and deselected actions are almost identical, we could use fallthrough in the switch statement. Only one line looks different: 
select - `newCategorySelected[action.categoryIndex] = true;`
deselect - `newCategorySelected[action.categoryIndex] = false;`

Reply:
Fall throughs are almost always (IMHO) a bad idea, as they save a bit of code but tend to surprise the reader unless they are paying close attention.